### DISCUSSION TITLE
How to accept only number in the flet textfield ?

### DISCUSSION THREAD

**Discussion by noob-infinity** [flet-dev/flet#1650] [Q&A]
Hi,

I want to know how to make a  textfield in flet  that accept only number . I have used attribute `keyboard_type=ft.KeyboardType.NUMBER`, however it is not working.

Please suggest.

Thanks in advance.

**Comment by niuniudd** [comment]
You can use:max_length=1

**Reply by noob-infinity** [reply]
First all,length of the input will vary much hence I don't want to set max length length.Even if set the max length,the textfield is still accepting non numeric character.

**Reply by niuniudd** [reply]
You can check  e.data in event method

**Comment by ndonkoHenri** [comment]
At the moment Flet doesn't have an ideal way to do this. There were several tries for this using `on_change` to filter the keyboard inputs. These tries weren't that ideal and the result wasn't that user friendly (one could see the typed non-numeric character being entered and later on removed).

Good to mention: the effect of `type` could be better observed on mobile. Mobile devices have different type of keyboards (one for normal text, one for numbers only, email...etc) and depending on the `type` value you set, the appropriate keyboard type is then shown.
